src/com/goide/intentions/GoMoveToStructInitializationIntention.java

```diff
@@ -61,11 +61,32 @@ private static Data getData(@NotNull PsiElement element) {
     if (!element.isValid() || !element.isWritable()) return null;
     GoAssignmentStatement assignment = getValidAssignmentParent(element);
     GoReferenceExpression selectedFieldReference = assignment != null ? getFieldReferenceExpression(element, assignment) : null;
-    GoCompositeLit compositeLit = selectedFieldReference != null ? getStructLiteralByReference(selectedFieldReference, assignment) : null;
-    if (compositeLit == null) return null;
+    if (selectedFieldReference == null) return null;
 
-    List<GoReferenceExpression> references = getUninitializedSingleFieldReferences(assignment, selectedFieldReference, compositeLit);
-    return !references.isEmpty() ? new Data(assignment, compositeLit, references) : null;
+    GoVarDefinition structDefinition = getDefinition(selectedFieldReference);
+    GoStatement previousStatement = PsiTreeUtil.getPrevSiblingOfType(assignment, GoStatement.class);
+    boolean needReplaceDeclarationWithShortVar = isUnassigned(getSingleVarSpecByDefinition(previousStatement, structDefinition));
+
+    GoCompositeLit compositeLit = needReplaceDeclarationWithShortVar
+                                  ? createStructLiteral(structDefinition, previousStatement.getProject())
```

Comment:
You shouldn't create struct literal at this place. Because method name is `getData` and it's called in `isAviable`